httpx/_client.py

```diff
@@ -1330,6 +1332,14 @@ def __init__(
             )
         self._mounts = dict(sorted(self._mounts.items()))
 
+    @property
+    def pending_requests(self) -> int:
+        return self._pending_requests
+
+    @pending_requests.setter
+    def pending_requests(self, p: int) -> None:
+        self._pending_requests = p
```

Comment:
I don't think we want to add any `client.pending_requests` attribute to the API. I'd suggest we keep any state here private.